### PR TITLE
Ensure status is not rendered

### PR DESCRIPTION
**Pull request by brancz**
When the `status` field is rendered, Kubernetes will validate it when trying to apply it, which causes errors as the `conditions` and `storedVersions` fields are validated and must be non-empty. However when rendering these we want to render them entirely without status anyways as they're just used to be applied against a Kubernetes cluster.

@ant31 @tamalsaha

### CODE REVIEW

File: pkg/cli-utils.go [hunk 1]
```diff
@@ -110,22 +110,47 @@ func NewCustomResourceDefinition(config Config) *extensionsobj.CustomResourceDef
 	return crd
 }
 
-func MarshallCrd(crd *extensionsobj.CustomResourceDefinition, outputFormat string) {
-	jsonBytes, err := json.MarshalIndent(crd, "", "    ")
+func MarshallCrd(crd *extensionsobj.CustomResourceDefinition, outputFormat string) error {
+	jsonBytes, err := json.Marshal(crd)
```
Comment (tamalsaha) [inline pkg/cli-utils.go:114]: Will this be simpler?

```
r := *crd
r.Status = CustomResourceDefinitionStatus{}
jsonBytes, err := json.MarshalIndent(r, "", "    ")
```

Comment (brancz) [reply pkg/cli-utils.go:114]: Unfortunately no, the CustomResourceDefinition object has a non pointer CustomResourceDefinitionStatus that has non pointer values, which always make it non-empty, therefore the omitempty is not triggered.

Comment (tamalsaha) [reply pkg/cli-utils.go:114]: I see what you mean. lgtm .

File: pkg/cli-utils.go [hunk 2]
```diff
@@ -110,22 +110,47 @@ func NewCustomResourceDefinition(config Config) *extensionsobj.CustomResourceDef
 	return crd
 }
 
-func MarshallCrd(crd *extensionsobj.CustomResourceDefinition, outputFormat string) {
-	jsonBytes, err := json.MarshalIndent(crd, "", "    ")
+func MarshallCrd(crd *extensionsobj.CustomResourceDefinition, outputFormat string) error {
```
Comment (ant31) [inline pkg/cli-utils.go:113]: the function signature is changing (return error), meaning it breaks all code importing it.

Comment (tamalsaha) [reply pkg/cli-utils.go:113]: I am ok with this, in case you folks are wondering.

Comment (brancz) [reply pkg/cli-utils.go:113]: I agree, it better to have this tiny breaking change (of an unversioned lib :) ) to handle errors appropirately.

**Pull request merged**